syntax = "proto3";
package personreid;
import "common.proto";
service PersonReIDAPI {
    rpc personReIDUpload (personReIDUploadRequest) returns (personReIDUploadResponse) {};
    rpc personReIDRealtime (personReIDRealtimeRequest) returns (personReIDRealtimeResponse) {};
    rpc personReIDHistory ( personReIDHistoryRequest) returns (personReIDHistoryResponse) {};
}
message personReIDUploadRequest {
    bytes rawdata = 1;   //上传待布控人员人体图片
    int32 Width = 2;
    int32 Hight = 3;
    int32 Channel = 4;
}

message personReIDUploadResponse {
    int32 status = 1;   // 0：有人体， 1：没有人体
}

message personReIDRealtimeRequest {
    bytes personEncodings = 1; // 人体特征
    bytes rawData = 2;  //待布控地点的实时图片（rawdata）
    int32 Width = 3;
    int32 Hight = 4;
    int32 Channel = 5;
}

message personReIDRealtimeResponse {
    repeated              int32 status = 1;     // 0:同一人； 1：不为同一人； 2：没有检测到行人
    repeated common.BoundingBox personRect = 2; // person bbox ；如果没有行人默认：[-1, -1, -1, -1]
}

message personReIDHistoryRequest {
   repeated string imagePath = 1;  // 人体路径列表[1, 2, 3, 4, 5 ...]
}

message personReIDHistoryResponse {
   repeated int32 index = 1; // 对应人体路径的状态列表[0, 1,0, 0, 0 ...]（其中1为是同一人，0为不是同一人）
}
